dids/diddht/diddht.go

```diff
@@ -1,175 +1,276 @@
+		if err != nil {
+			return did.BearerDID{}, fmt.Errorf("failed to get public key for verification method: %w", err)
 		}
 
-		name := h.Name.String()
-		fullTxtRecord := strings.Join(value.TXT, "")
-		if strings.HasPrefix(name, "_did") {
-			didRecord.rootRecord = fullTxtRecord
-			continue
+		vmPublicKeyBytes, err := dsa.PublicKeyToBytes(vmPublicKey)
+		if err != nil {
+			return did.BearerDID{}, fmt.Errorf("failed to convert public key to bytes for verification method: %w", err)
 		}
 
-		if _, ok := didRecord.records[h.Name.String()]; !ok {
-			fmt.Println("TODO semantic description")
+		controller := func() string {
+			if pk.controller != "" {
+				return pk.controller
+			}
+
+			return bdid.ID
+		}()
+
+		vmZbase32Encoded := zbase32.EncodeToString(vmPublicKeyBytes)
+		newVM := didcore.VerificationMethod{
+			ID:           "did:dht:" + vmZbase32Encoded + "#" + vmKeyID,
+			Type:         "JsonWebKey",
+			Controller:   controller,
+			PublicKeyJwk: &vmPublicKey,
 		}
-		didRecord.records[h.Name.String()] = fullTxtRecord
+
+		document.AddVerificationMethod(newVM, didcore.Purposes(pk.purposes...))
 	}
 
-	return &didRecord, nil
-}
+	for _, service := range o.services {
+		s := service
+		document.AddService(&s)
+	}
 
-// TODO on the diddhtrecord we should validate the minimum reqs for a valid did
-func parseVerificationRelationships(rootRecord string) (map[string][]string, error) {
-	rootRecordProps, err := parseTXTRecordData(rootRecord)
+	// 5. Map the output DID Document to a DNS packet
+	msgBytes, err := dns.MarshalDIDDocument(&document)
 	if err != nil {
-		return nil, err
-	}
-	// reverse the map to get the relationships
-	relationshipMap := map[string][]string{}
-	for k, values := range rootRecordProps {
-		v := strings.Join(values, "")
-		rel, ok := relationshipMap[v]
-		if !ok {
-			rel = []string{}
-		}
-		rel = append(rel, k)
-		relationshipMap[v] = rel
+		return did.BearerDID{}, fmt.Errorf("failed to marshal did document to dns packet: %w", err)
 	}
 
-	return relationshipMap, nil
-}
+	// 6. Construct a signed BEP44 put message with the v value as a bencoded DNS packet from the prior step.
+	seq := time.Now().Unix() / 1000
 
-func parseTXTRecordData(data string) (map[string][]string, error) {
-	result := map[string][]string{}
-	fields := strings.Split(data, ";")
-	if len(fields) == 0 {
-		return nil, errors.New("no fields found")
+	signer := func(payload []byte) ([]byte, error) {
+		return keyMgr.Sign(keyID, payload)
 	}
-	for _, field := range fields {
-		kv := strings.Split(field, "=")
-		if len(kv) != 2 {
-			return nil, fmt.Errorf("malformed field %s", field)
-		}
-		k, v := kv[0], strings.Split(kv[1], ",")
-		current, ok := result[k]
-		if ok {
-			v = append(current, v...)
-		}
-		result[k] = v
+
+	bep44Msg, err := bep44.NewMessage(msgBytes, seq, publicKeyBytes, signer)
+	if err != nil {
+		return did.BearerDID{}, fmt.Errorf("failed to create signed bep44 message: %w", err)
+	}
```

Comment:
there can be a check here to make sure the payload <= 1000 bytes

Reply:
adding as part of the `NewMessage`